olp-cpp-sdk-dataservice-read/src/generated/api/StreamApi.h

```diff
@@ -0,0 +1,119 @@
+   *
+   * @param client Instance of OlpClient used to make REST request.
+   * @param layer_id The ID of the stream layer.
+   * @param subscription_id Include this parameter if you want to look up the
+   * `nodeBaseURL` for a given subscriptionId.
+   * @param mode Specifies whether to use serial or parallel subscription mode.
+   * For more details see [Get Data from a Stream
+   * Layer](https://developer.here.com/olp/documentation/data-store/data_dev_guide/rest/getting-data-stream.html).
+   * @param consumer_id The ID to use to identify this consumer. It must be
+   * unique within the consumer group. If you do not provide one, the system
+   * will generate one.
+   * @param subscription_properties One or more Kafka Consumer properties to use
+   * for this subscription. For more information, see [Get Data from a Stream
+   * Layer](https://developer.here.com/olp/documentation/data-store/data_dev_guide/rest/getting-data-stream.html).
+   * @param context A CancellationContext, which can be used to cancel the
+   * pending request.
+   * @return The result of operation as a client::ApiResponse object.
+   */
+  static SubscribeApiResponse Subscribe(
+      const client::OlpClient& client, const std::string& layer_id,
+      const boost::optional<std::string>& subscription_id,
+      const boost::optional<std::string>& mode,
+      const boost::optional<std::string>& consumer_id,
+      const boost::optional<ConsumerProperties>& subscription_properties,
+      const client::CancellationContext& context,
+      std::string& x_correlation_id /*out*/);
+
+  /**
+   * @brief Delete subscription to a layer.
+   *
+   * This operation removes the subscription from the service. The base path to
+   * use is the value of 'nodeBaseURL' returned from /subscribe POST request.
+   *
+   * @param client Instance of OlpClient used to make REST request.
+   * @param layer_id The ID of the stream layer.
+   * @param subscription_id The subscriptionId received in the response of the
+   * /subscribe request (required if mode=parallel).
+   * @param mode The subscription mode of this subscriptionId (as provided in
+   * /subscribe POST API).
+   * @param x_correlation_id A trace ID to use to associate this request with
+   * other requests in your process. The correlation ID is the value of the
+   * response header `X-Correlation-Id` from the prior request in your process.
+   * Once you use a correlation ID in a `/delete` request, do not use it again
+   * since `/delete` marks the end of a process.
+   * @param context A CancellationContext, which can be used to cancel the
+   * pending request.
+   * @return The result of operation as a client::ApiResponse object.
+   */
+  static DeleteSubscriptionApiResponse DeleteSubscription(
+      const client::OlpClient& client, const std::string& layer_id,
+      const boost::optional<std::string>& subscription_id,
```

Comment:
You cannot delete subscription without providing the subscription_id and mode correctly.
You will receive a strange error back saying subscription not found.
Means subscription_id and mode should be mandatory.